### PR TITLE
V1.3 Update Docs for Filter feature

### PR DESCRIPTION
**Pull request by jyztintan**
# Description:
This PR adds documentation for the new filter by name and filter by remarks functionality. 

# Key Changes:
- Include UML Diagram for Filter Command
- Update Developer Guide for Filter command (Implementation, Example usage, Alternatives considered)

### CODE REVIEW

**Comment by jyztintan** [comment]
Addresses (but not fix) #139 and #115 for filter command

**Review by appleraincoat — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -162,6 +162,77 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+The Filter by Name feature allows users to filter the list of persons in Realodex based on their names. 
+It is implemented using a predicate that checks if a person's name contains the keyphrase provided by the user. 
+The key components involved in this feature include:
+
+- FilterCommand: A command that, when executed, updates the filtered person list based on the predicate.
+- NameContainsKeyphrasePredicate: A predicate that tests whether a person's name contains the given keyphrase.
+
+#### Example Usage Scenario
+1. The user launches the application and the Model is initialized with the initial list of persons.
+2.  The user executes the command filter n/John, intending to filter out persons whose names contain "John".
```
Comment (appleraincoat) [inline docs/DeveloperGuide.md:178]: extra space here

Comment (jyztintan) [reply docs/DeveloperGuide.md]: resolved thanks

**Pull request merged**